2019/05/5.rb

```diff
@@ -0,0 +1,169 @@
+    second_input_value = second_input_index
+  else
+    second_input_value = Integer(intcode[second_input_index])
+  end
+  
+  return second_input_value
+end
+
+def get_third_parameter_value(command, intcode, index)
+  output_index = Integer(intcode[index+3])
+end
+
+def run_intcode(input)
+  
+  i = 0 
+  loop do
+  
+    if i == input.length
+      break
+    end
+    
+    command = input[i]
+    opcode = get_opcode_for_command(command.to_s)
+       
+    if opcode == :halt
+      
+      puts "End Program"
+      break
+      
+    elsif opcode == :add
+        
+      first_input_value = get_first_parameter_value(command, input, i)
+      second_input_value = get_second_parameter_value(command, input, i)   
+      output_index = get_third_parameter_value(command, input, i)
+      
+      result = first_input_value + second_input_value
+      input[output_index] = result.to_s
+      
+    elsif opcode == :multiply
+
+      first_input_value = get_first_parameter_value(command, input, i)
+      second_input_value = get_second_parameter_value(command, input, i)
+      output_index = get_third_parameter_value(command, input, i)
+
+      result = first_input_value * second_input_value
+      input[output_index] = result.to_s
+
+    elsif opcode == :read
+
+      first_input_value = Integer(input[i+1])
+      input[first_input_value] = 5 # first input value, this is a dirty hack
+      
+    elsif opcode == :write
+    
+      first_input_index = Integer(input[i+1])
+      puts input[first_input_index]
+
+    elsif opcode == :jump_if_true
+      first_input_value = get_first_parameter_value(command, input, i)
+      second_input_value = get_second_parameter_value(command, input, i)
+
+      if first_input_value != 0
+        i = Integer(second_input_value)
+        next
+      end
+
+    elsif opcode == :jump_if_false
+      first_input_value = get_first_parameter_value(command, input, i)
+      second_input_value = get_second_parameter_value(command, input, i)
+
+      if first_input_value == 0
+        i = Integer(second_input_value)
+        next
+      end
+
+    elsif opcode == :less_than
+      first_input_value = get_first_parameter_value(command, input, i)
+      second_input_value = get_second_parameter_value(command, input, i)
+      output_index = Integer(input[i+3])
+      
+      if first_input_value < second_input_value
+        input[output_index] = "1"
+      else
+        input[output_index] = "0"
+      end
+
+    elsif opcode == :equals
+      first_input_value = get_first_parameter_value(command, input, i)
+      second_input_value = get_second_parameter_value(command, input, i)
+      output_index = Integer(input[i+3])
+      
+      parameters_equal = first_input_value == second_input_value
+      if parameters_equal
+      	input[output_index] = "1"
+      else
+        input[output_index] = "0"
+      end
+    else
+      puts "Weird opcode (#{opcode} at #{i}), command: #{command}, aborting..."
```

Comment:
You're not really aborting 😉 

```suggestion
      raise "Weird opcode (#{opcode} at #{i}), command: #{command}, aborting..."
```